lib/binding/kvdb_interface.c

```diff
@@ -1140,20 +1140,67 @@ hse_kvs_cursor_read(
 {
     merr_t err;
 
-    if (HSE_UNLIKELY(!cursor || !key || !klen || !val || !vlen || !eof || flags != 0))
+    if (HSE_UNLIKELY(!cursor || !key || !klen || !eof || flags != 0))
+        return merr(EINVAL);
+
+    if (HSE_UNLIKELY(!!val ^ !!vlen))
         return merr(EINVAL);
```

Comment:
We could be less restrictive and define semantics as follows:
- read value if and only if `val != NULL and vlen != NULL and *vlen > 0`
- it is not an error if val or vlen are NULL
Thoughts?

Reply:
Still in the camp of vlen doesn't matter if HSE doesn't require it!

Reply:
Will fix in another PR